cmd/segment-reaper/observer.go

```diff
@@ -143,11 +145,88 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
+				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
+					// TODO should we also check if its valid sequence
+					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+					brokenObject = true
+				}
+
+				if brokenObject {
+					err := obsvr.printSegment(ctx, "l", bucket, path, obsvr.writer)
```

Comment:
is the argument order for `"l"` and `bucket` correct?

Reply:
Because of this bug and what @kaloyan-raev commented in a previous review, we could change the signature of the `printSegment` to something like 

```
printSegment(ctx context.Context, bucket string, segmentIndex int, isLastSegment bool, path storj.Path, csvWriter *csv.Writer) error
```

And doing something similar for `pointerCreationDate` function.

Reply:
I'm not 100% convinced that this will be cleaner. If we will have `segmentIndex` as `int` then we need to handle `-1` in both methods: `printSegment` and `pointerCreationDate`.

Reply:
I'm OK leaving it as it's.